fw_django/shopapp/models.py

```diff
@@ -0,0 +1,49 @@
+from django.core.validators import RegexValidator
+from django.db import models
+
+
+class Client(models.Model):
+    name = models.CharField(max_length=100)
+    email = models.EmailField()
+    phone_regex = RegexValidator(regex=r'^\+?1?\d{9,15}$',
+                                 message="Phone number must be entered in the format: "
+                                         "'+0123456789'. Up to 15 digits allowed.")
+    phone_number = models.CharField(validators=[phone_regex], max_length=17, blank=True)
+    reg_date = models.DateField(auto_now_add=True)
+
+    def __str__(self):
+        return f'{self.name} {self.email}'
+
+
+class Product(models.Model):
+    name = models.CharField(max_length=100)
+    description = models.TextField()
+    price = models.DecimalField(max_digits=10, decimal_places=2)
+    stock = models.PositiveIntegerField(default=0)
+
+    def __str__(self):
+        return f'{self.name}'
+
+
+class Order(models.Model):
+    owner = models.ForeignKey(Client, related_name='owner_order_set', on_delete=models.CASCADE)
+    created = models.DateField(auto_now_add=True)
+
+    def __str__(self):
+        return f'Order {self.id}'
+
+    def get_total_cost(self):
+        return sum(item.get_cost() for item in self.items.all())
```

Comment:
Должно быть как-то так
```suggestion
class Order(models.Model):
    owner= models.ForeignKey(Client, related_name='owner_order_set', on_delete=models.CASCADE)
    product = models.ManyToManyField(Product)
    total_price = models.DecimalField(max_digits=8, decimal_places=2)
    created = models.DateTimeField(auto_now_add=True)
```
И все

Reply:
Как определять количество одного продукта в заказе с такой моделью?